### DISCUSSION TITLE
Should cupsGetDestWithURI support http/https?

### DISCUSSION THREAD

**Discussion by xiaowen581** [OpenPrinting/cups#979] [Ideas] [answered]
Since CUPS has the ability to print to a Windows print server, would it be better to have cupsGetDestWithURI support http/https? 

Now, we have to convert the http printer URI to the ipp one before calling cupsGetDestWithURI... 
For example: 
http://1.2.3.4/printers/RICOH/.printer => ipp://1.2.3.4:80/printers/RICOH/.printer

**Comment by michaelrsweet** [comment, marked as answer]
Using "http" or "https" URIs for IPP was always experimental. IPP/1.1 and higher requires support for "ipp" and "ipps", respectively. CUPS 2.5 has dropped the http/https backend symlinks as well.

But more importantly, the cupsGetDestWithURI support is for printers that support IPP Everywhere, Mopria, or AirPrint (IPP/2.0 plus standard document formats and at least the basic IPP Job attributes) - the old Windows IPP/1.0 support basically acted as a LPD replacement and not a driver replacement.

So I would say no, we don't want to support http/https URIs.
